include/substrait/function/Function.h

```diff
@@ -0,0 +1,119 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "substrait/function/FunctionSignature.h"
+#include "substrait/type/Type.h"
+
+namespace io::substrait {
+
+struct FunctionArgument {
+  [[nodiscard]] virtual bool isRequired() const = 0;
+
+  /// Convert argument type to short type string based on
+  /// https://substrait.io/extensions/#function-signature-compound-names
+  [[nodiscard]] virtual std::string toTypeString() const = 0;
+
+  virtual bool isWildcardType() const {
+    return false;
+  };
+
+  [[nodiscard]] virtual bool isValueArgument() const {
```

Comment:
Why `isValueArgument` and not `isEnumArgument` or `isTypeArgument`?

Reply:
updated